elasticsearch/src/main/java/com/gentics/mesh/search/index/node/NodeContainerTransformer.java

```diff
@@ -413,7 +414,22 @@ public JsonObject toPermissionPartial(Node node, ContainerType type) {
 	}
 
 	public String generateVersion(NodeGraphFieldContainer container, String releaseUuid, ContainerType type) {
-		return ETag.hash(toDocument(container, releaseUuid, type, false).encode());
+		Node node = container.getParentNode();
+		Project project = node.getProject();
+
+		StringBuilder builder = new StringBuilder();
+		builder.append(container.getElementVersion());
+		builder.append("|");
+		builder.append(releaseUuid);
+		builder.append("|");
+		builder.append(type.name());
+		builder.append("|");
+		builder.append(project.getElementVersion());
+		builder.append("|");
+		builder.append(node.getElementVersion());
+
+		String hash = ETag.hash(builder.toString());
+		return hash;
 	}
```

Comment:
Consider Stream.of(...).collect(Collectors.joining("|"))

Reply:
Consider it. Rejected